lib/Sema/TypeCheckPattern.cpp

```diff
@@ -1024,7 +1025,7 @@ bool TypeChecker::coercePatternToType(Pattern *&P, DeclContext *dc, Type type,
     // If this is the payload of an enum, and the type is a single-element
     // labeled tuple, treat this as a tuple pattern. It's unlikely that the
     // user is interested in binding a variable of type (foo: Int).
-    if ((options & TypeResolutionFlags::EnumPatternPayload)
+    if ((options.getContext() == TypeResolverContext::EnumPatternPayload)
```

Comment:
maybe options.is(TypeResolverContext::EnumPatternPayload) would read better?

Reply:
Sure. Just FYI – that won't remove the need for `getContext()` thought (for use with `switch` statements, etc).